### PR TITLE
feat: add builder esbuild plugin doc

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: website/builder/src/zh/plugins/plugin-esbuild.md [hunk 1]
```diff
@@ -1 +1,92 @@
...
+const DEFAULT_MINIFY_OPTIONS = {
+  target: 'es2015',
+};
+```
+
+如果你需要修改压缩参数，可以查看[官方文档](https://github.com/privatenumber/esbuild-loader#minifyplugin)。
+
+## 限制条件
+
+虽然 ESBuild/SWC 能给现有的 Webpack 项目带来明显的构建性能提升，但这两个工具在接入 Eden 时还存在一定的局限性，需要大家在接入的时候格外注意。
```
Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:68]: Eden -> Builder

Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:68]: Webpack -> webpack (官方推荐小写）

File: website/builder/src/zh/plugins/plugin-esbuild.md [hunk 2]
```diff
@@ -1 +1,92 @@
...
+
+### ESBuild 限制
+
+作为代码转译工具(即 `loader` 能力)，ESBuild 仅能支持到 ES2015(即 ES6) 语法，如果生产环境需要降级到 ES5 及以下的语法，**请在生产环境关闭 ESBuild**。另外，由于 Eden 底层使用的是 ESBuild 的`Transform API`，并不支持 ESBuild 插件来进行自定义编译过程，因此，诸如 `babel-plugin-import` 等原有 Babel 插件的语法编译功能在开启 ESBuild 后无法使用！
+
+作为代码压缩工具(即 `minimize` 能力)，ESBuild 可以在生产环境中进行压缩和混淆，但也仅能支持 ES2015 即以上的语法(因为压缩过程需要识别代码 AST 并进行语法转换)。你可以通过如下的配置指定目标语法版本:
+
+```ts
+import { createEdenConfig } from '@ies/eden-web-build';
```
Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:77]: eden 相关示例..

Comment (sanyuan0704) [reply website/builder/src/zh/plugins/plugin-esbuild.md:77]: sorry, 改动忘记同步给中文文档了

File: website/builder/src/zh/plugins/plugin-esbuild.md [hunk 3]
```diff
@@ -1 +1,86 @@
...
+
+如果你需要修改压缩参数，可以查看[官方文档](https://github.com/privatenumber/esbuild-loader#minifyplugin)。
+
+## 限制条件
+
+虽然 ESBuild 能给现有的 webpack 项目带来明显的构建性能提升，但这两个工具在接入 Builder 时还存在一定的局限性，需要大家在接入的时候格外注意。
+
+### ESBuild 限制
+
+作为代码转译工具(即 `loader` 能力)，ESBuild 仅能支持到 ES2015(即 ES6) 语法，如果生产环境需要降级到 ES5 及以下的语法，**请在生产环境关闭 ESBuild**。另外，由于 Eden 底层使用的是 ESBuild 的`Transform API`，并不支持 ESBuild 插件来进行自定义编译过程，因此，诸如 `babel-plugin-import` 等原有 Babel 插件的语法编译功能在开启 ESBuild 后无法使用！
```
Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:72]: 这里还有一个 eden 哈

Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:72]: 不太推荐用感叹号，语气太强了

File: website/builder/src/zh/plugins/plugin-esbuild.md [hunk 4]
```diff
@@ -1 +1,86 @@
...
+
+作为代码压缩工具(即 `minimize` 能力)，ESBuild 可以在生产环境中进行压缩和混淆，但也仅能支持 ES2015 即以上的语法(因为压缩过程需要识别代码 AST 并进行语法转换)。你可以通过如下的配置指定目标语法版本:
```
Comment (chenjiahan) [inline website/builder/src/zh/plugins/plugin-esbuild.md:74]: 即以上 -> 及以上

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**